pkg/prometheus/promcfg.go

```diff
@@ -62,13 +62,13 @@ type ConfigGenerator struct {
 	version                semver.Version
 	notCompatible          bool
 	prom                   monitoringv1.PrometheusInterface
-	endpointSliceSupported bool
+	useEndpointSlice       bool // Whether to use EndpointSlice for service discovery from `ServiceMonitor` objects.
 	scrapeClasses          map[string]monitoringv1.ScrapeClass
 	defaultScrapeClassName string
 }
 
 // NewConfigGenerator creates a ConfigGenerator for the provided Prometheus resource.
-func NewConfigGenerator(logger log.Logger, p monitoringv1.PrometheusInterface, endpointSliceSupported bool) (*ConfigGenerator, error) {
+func NewConfigGenerator(logger log.Logger, p monitoringv1.PrometheusInterface) (*ConfigGenerator, error) {
```

Comment:
Do I understand correctly that you're removing the argument that tells if the k8s cluster where the operator is running supports EndpointSlices or not? Shouldn't this information be kept and used?

E.g. what happens if the user sets the role to EndpointSlice, but the cluster doesn't support it?

Reply:
Thank you for pointing this out. It was an oversight on my part. Your comment [here](https://github.com/prometheus-operator/prometheus-operator/pull/6672#discussion_r1662943437) helped me realize the mistake I made.

I have re-added the argument to pass through the support of endpointslices here [16dc1f3](https://github.com/prometheus-operator/prometheus-operator/pull/6672/commits/16dc1f3d8a62d9076512e7e47993d7153177c4a9)